SuccessEycaActivationActivity/eyca.ts

```diff
@@ -0,0 +1,139 @@
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { format } from "date-fns";
+import { toError } from "fp-ts/lib/Either";
+import {
+  fromEither,
+  fromLeft,
+  taskEither,
+  tryCatch
+} from "fp-ts/lib/TaskEither";
+import { RedisClient } from "redis";
+import { EycaAPIClient } from "../clients/eyca";
+import { Timestamp } from "../generated/definitions/Timestamp";
+import { CcdbNumber } from "../generated/eyca-api/CcdbNumber";
+import { ErrorResponse } from "../generated/eyca-api/ErrorResponse";
+import { errorsToError } from "../utils/conversions";
+import { getTask, setWithExpirationTask } from "../utils/redis_storage";
+
+export const CCDB_SESSION_ID_KEY = "CCDB_SESSION_ID";
+export const CCDB_SESSION_ID_TTL = 1500;
+
+const ccdbLogin = (
+  eycaClient: ReturnType<EycaAPIClient>,
+  username: NonEmptyString,
+  password: NonEmptyString
+) =>
+  tryCatch(
+    () =>
+      eycaClient.authLogin({
+        password,
+        username
+      }),
+    toError
+  )
+    .mapLeft(err => new Error(`Cannot call EYCA authLogin API ${err.message}`))
+    .chain(_ => fromEither(_).mapLeft(errorsToError))
+    .chain<NonEmptyString>(res =>
+      res.status !== 200 || ErrorResponse.is(res.value.api_response)
+        ? fromLeft(
+            new Error(
+              `Error on EYCA authLogin API|STATUS=${res.status}, DETAIL=${res.value.api_response.text}`
+            )
+          )
+        : taskEither.of(res.value.api_response.text)
+    );
+
+const retrieveCcdbSessionId = (
+  redisClient: RedisClient,
+  eycaClient: ReturnType<EycaAPIClient>,
+  username: NonEmptyString,
+  password: NonEmptyString
+) =>
+  getTask(redisClient, CCDB_SESSION_ID_KEY).foldTaskEither(
+    () => ccdbLogin(eycaClient, username, password),
```

Comment:
Are we using `ccdbLogin` as fallback on Redis errors? We can only return an error, `retrieveCcdbSessionId` is used inside `SuccessEycaActivationActivity` that has a retry logic.

Reply:
Yep I will restore the default behaviour. So we let the activity to retry and log errors in case Redis get or set fails.